plugins/innogrit/innogrit-nvme.c

```diff
@@ -0,0 +1,534 @@
+	printf("DW7       Min Ec Cnt                    : %u\n", pvsc_smart->min_ec_cnt);
+	printf("DW8       Avg Ec Cnt                    : %u\n", pvsc_smart->avg_ec_cnt);
+	printf("DW9       Mrd Rr Good Cnt               : %u\n", pvsc_smart->mrd_rr_good_cnt);
+	printf("DW10      Ard Rr Good Cnt               : %u\n", pvsc_smart->ard_rr_good_cnt);
+	printf("DW11      Preset Cnt                    : %u\n", pvsc_smart->preset_cnt);
+	printf("DW12      Nvme Reset Cnt                : %u\n", pvsc_smart->nvme_reset_cnt);
+	printf("DW13      Low Pwr Cnt                   : %u\n", pvsc_smart->low_pwr_cnt);
+	printf("DW14      Wa                            : %u\n", pvsc_smart->wa);
+	printf("DW15      Ps3 Entry Cnt                 : %u\n", pvsc_smart->ps3_entry_cnt);
+	printf("DW16[0]   highest_temp[0]               : %u\n", pvsc_smart->highest_temp[0]); 
+	printf("DW16[1]   highest_temp[1]               : %u\n", pvsc_smart->highest_temp[1]); 
+	printf("DW16[2]   highest_temp[2]               : %u\n", pvsc_smart->highest_temp[2]); 
+	printf("DW16[3]   highest_temp[3]               : %u\n", pvsc_smart->highest_temp[3]); 
+	printf("DW17      weight_ec                     : %u\n", pvsc_smart->weight_ec);
+	printf("DW18      slc_cap_mb                    : %u\n", pvsc_smart->slc_cap_mb);
+	printf("DW19-20   nand_page_write_cnt           : %llu\n", pvsc_smart->nand_page_write_cnt);
+
+	iIndex = 21;
+	for(i = 0; i < (sizeof(pvsc_smart->reserved2) /4); i++)
+	{
+		if(pvsc_smart->reserved2[i] != 0)
+			printf("DW%-2d                                    : %u\n", iIndex, pvsc_smart->reserved2[i]);
+		iIndex++;
+	}
+
+	return 0;
+}
+
+
+
+
+/*****************************************************************************
+*
+* ENTRY POINT: SortEventLogFn
+*
+*****************************************************************************/
+int SortEventLogFn(const void * a, const void * b)
+{
+	int rc = 0;
+	if (((PEVENTLOG_ADDINDEX)a)->ms > ((PEVENTLOG_ADDINDEX)b)->ms)
+	{
+		rc = 1;
+	}
+	else if (((PEVENTLOG_ADDINDEX)a)->ms < ((PEVENTLOG_ADDINDEX)b)->ms)
+	{
+		rc = -1;
+	}
+	else
+	{		
+		if( ((PEVENTLOG_ADDINDEX)a)->iIndex < ((PEVENTLOG_ADDINDEX)b)->iIndex) 
+			rc = -1;
+		else
+			rc = 1;
+	}
+
+	return  rc;
+}
+
+
+
+
+
+/*****************************************************************************
+*
+* ENTRY POINT: SortEventLog
+*
+*****************************************************************************/
+static void SortEventLog(PEVENTLOG Data16KSrc, unsigned int iCount)
+{	
+	EVENTLOG_ADDINDEX pEventLogAdd[512];
+	unsigned int i;
+	
+	// iCount = iDataLen / sizeof(EVENTLOG) - 1;		
+	for(i = 0; i < iCount; i++)
+	{
+		memcpy(&pEventLogAdd[i], &Data16KSrc[i], sizeof(EVENTLOG));
+		pEventLogAdd[i].iIndex = i;
+	}
+			
+	qsort(pEventLogAdd, iCount, sizeof(EVENTLOG_ADDINDEX), SortEventLogFn);
+	
+	
+	for(i = 0; i < iCount; i++)
+	{
+		memcpy(&Data16KSrc[i], &pEventLogAdd[i], sizeof(EVENTLOG));
+	}	
+}
```

Comment:
There are so many whitespace and formatting issue in this function, see the red
markings:

![image](https://user-images.githubusercontent.com/1050803/173365669-514c1416-d4b8-4929-96d2-5b86036581d1.png)